docs/src/markdown/documentation/kotlin/building-a-workflow.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Building a Workflow
+index: 2
+navigation:
+    visible: true
+    path: documentation
+---
+
+# Building a Workflow
+
+
+## Introduction
+
+A simple workflow looks something like this:
+
+```kotlin
+class DemoWorkflow : Workflow<String, Unit, Nothing, String> {
+
+    override fun initialState(input: String) = Unit
+
+    override fun compose(
+        input: String,
+        state: Unit,
+        context: WorkflowContext<Unit, Nothing>
+    ) = "Hello, $name"
+
+    override fun snapshotState(state: Unit) = Snapshot.EMPTY
+    override fun restoreState(snapshot: Snapshot) = Unit
+}
+```
+
+A class implementing `Workflow` represents a single node in the workflow tree. It may store resources like network services as properties (e.g. provided via dependency injection). Configuration values that may change over the lifetime of the workflow should be provided through the `InputT` type (a simple `String` in the example above).
+
+Every workflow defines its own `StateT` type to contain any data that should persist through subsequent compose passes.
+
+## Compose
+
+Workflows are only useful when they render a value for use by their parent (or, if they are the root workflow, for display). This type is very commonly a view model, or `Screen`. The `compose(state:context:)` method has a couple of parameters, so we’ll work through them one by one.
+
+```swift
```

Comment:
Just passing by. Since there is a lot of Swift code and the library targets both platforms maybe it makes sense to have the same text but different code samples. [Like Google does for Java and Kotlin](https://developer.android.com/guide/topics/ui/ui-events).